libs/ledger-live-common/src/exchange/providers/swap.ts

```diff
@@ -176,16 +168,13 @@ export const fetchAndMergeProviderData = async () => {
       getProvidersCDNData(),
     ]);
 
-    const transformedProvidersData = transformData(providersData);
-    const finalProvidersData = mergeProviderData(transformedProvidersData, providersExtraData);
+    const finalProvidersData = mergeProviderData(providersData, providersExtraData);
     providerDataCache = finalProvidersData;
 
     return finalProvidersData;
   } catch (error) {
     console.error("Error fetching or processing provider data:", error);
-    const transformedProvidersData = transformData(swapProviders);
-    const finalProvidersData = mergeProviderData(transformedProvidersData, swapAdditionData);
-    providerDataCache = finalProvidersData;
+    const finalProvidersData = mergeProviderData(swapProviders, swapAdditionData);
```

Comment:
readability : I would rename variables to tell which are the `staticDefaultSwapProviders` and which are the `cdnFetchedSwapProviders`, same for `staticDefaultSwapProvidersCalData` and `calFetchedSwapProvidersCalData`